presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveLocationService.java

```diff
@@ -0,0 +1,119 @@
+
+public class TestHiveLocationService
+{
+    @Test
+    public void testGetTableWriteInfo_append()
+    {
+        assertThat(locationHandle(LocationHandle.WriteMode.STAGE_AND_MOVE_TO_TARGET_DIRECTORY), false)
+                .produceWriteInfoAs(new LocationService.WriteInfo(
+                        new Path("hdfs://dir001/target"),
+                        new Path("hdfs://dir001/write"),
+                        LocationHandle.WriteMode.STAGE_AND_MOVE_TO_TARGET_DIRECTORY));
+
+        assertThat(locationHandle(
+                LocationHandle.WriteMode.DIRECT_TO_TARGET_EXISTING_DIRECTORY,
+                "hdfs://dir001/target",
+                "hdfs://dir001/target"),
+                false)
+                .produceWriteInfoAs(new LocationService.WriteInfo(
+                        new Path("hdfs://dir001/target"),
+                        new Path("hdfs://dir001/target"),
+                        LocationHandle.WriteMode.DIRECT_TO_TARGET_EXISTING_DIRECTORY));
+
+        assertThat(locationHandle(
+                LocationHandle.WriteMode.DIRECT_TO_TARGET_NEW_DIRECTORY,
+                "hdfs://dir001/target",
+                "hdfs://dir001/target"),
+                false)
+                .produceWriteInfoAs(new LocationService.WriteInfo(
+                        new Path("hdfs://dir001/target"),
+                        new Path("hdfs://dir001/target"),
+                        LocationHandle.WriteMode.DIRECT_TO_TARGET_NEW_DIRECTORY));
+    }
+
+    @Test
+    public void testGetTableWriteInfo_overwriteSuccess()
+    {
+        assertThat(locationHandle(LocationHandle.WriteMode.STAGE_AND_MOVE_TO_TARGET_DIRECTORY), true)
+                .produceWriteInfoAs(new LocationService.WriteInfo(
+                        new Path("hdfs://dir001/target"),
+                        new Path("hdfs://dir001/write"),
+                        LocationHandle.WriteMode.STAGE_AND_MOVE_TO_TARGET_DIRECTORY));
+    }
+
+    @Test(expectedExceptions = UnsupportedOperationException.class)
+    public void testGetTableWriteInfo_overwriteFail_1()
+    {
+        assertThat(locationHandle(LocationHandle.WriteMode.DIRECT_TO_TARGET_NEW_DIRECTORY,
+                "hdfs://dir001/target", "hdfs://dir001/target"), true);
+    }
+
+    @Test(expectedExceptions = UnsupportedOperationException.class)
+    public void testGetTableWriteInfo_overwriteFail_2()
+    {
+        assertThat(locationHandle(LocationHandle.WriteMode.DIRECT_TO_TARGET_EXISTING_DIRECTORY,
+                "hdfs://dir001/target", "hdfs://dir001/target"), true);
+    }
+
+    private static Assertion assertThat(LocationHandle locationHandle, boolean overwrite)
+    {
+        return new Assertion(locationHandle, overwrite);
+    }
+
+    public static class Assertion
+    {
+        private LocationService.WriteInfo actual;
+
+        public Assertion(LocationHandle locationHandle, boolean overwrite)
+        {
+            HdfsEnvironment hdfsEnvironment = mock(HdfsEnvironment.class);
```

Comment:
Rather than `mock()`, reuse `TestingHdfsEnvironment` in `TestBackgroundHiveSplitLoader` (make it public)